### ISSUE TITLE
Following breach.cc/hack instructions does not work

### ISSUE THREAD

**Issue by AdrianTP** [spolu/breach_core#110]
I am unable to install my local copy of mod_strip, and reinstalling the mod_strip that came with Breach gives me errors when running it:

```
OPTION: --no-chrome
module.js:333
    throw err;
          ^
Error: Cannot find module 'debug'
    at Function.Module._resolveFilename (module.js:331:15)
    at Function.Module._load (module.js:273:25)
    at Module.require (module.js:357:17)
    at require (module.js:373:17)
    at Object.<anonymous> (/Users/<username>/Library/Application Support/breach/modules/breach/mod_strip#v0.2.9-alpha.5/node_modules/express/node_modules/finalhandler/index.js:11:13)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:349:32)
    at Function.Module._load (module.js:305:12)
    at Module.require (module.js:357:17)
```

I'm not sure where to go from here...

**Comment by miketheprogrammer** [comment]
@AdrianTP In my experience some of the precached modules like express dont have all the dependencies correctly installs. please cd into the directory and start npm installing 1 by 1. It worked for me.

**Comment by AdrianTP** [comment]
That's odd that the official install of mod_strip didn't install all its dependencies automatically; that's supposed to chain, according to my understanding. I did get some `package.json` warnings in the mod_strip directory:

```
mod_strip#v0.2.9-alpha.5 <username>$ npm install
npm WARN package.json mod_strip@0.2.9-alpha.5 No description
npm WARN package.json mod_strip@0.2.9-alpha.5 No repository field.
npm WARN package.json mod_strip@0.2.9-alpha.5 No README data
```

Those shouldn't affect the installation, though...

That said, all I had to do was `rm` the `node_modules` directory in the (Application Support) mod_strip directory and redo `npm install` manually, and it cascaded all the dependencies just fine on its own. Thank you for the suggestion. :+1: 

Now how do I get it to install my local git clone of mod_strip? I enter `local:~/mod_strip` and click the `install` link but nothing happens (it just clears the input field and ignores me).

Edit: Actually, it gives me an error in the dev tools:

```
Failed to load resource: the server responded with a status of 500 (Internal Server Error) http://127.0.0.1:53524/modules/add
```

Edit again: Running a POST to the above URL from Postman in Chrome gives the following error in the response body:

```
Error: Invalid module `path`: undefined
    at Object.exports.err (/Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/lib/common.js:314:13)
    at augment_path (/Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/lib/module_manager.js:261:25)
    at EventEmitter.add (/Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/lib/module_manager.js:724:5)
    at /Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/lib/core_ui.js:236:41
    at fn (/Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/node_modules/async/lib/async.js:641:34)
    at Immediate._onImmediate (/Users/<username>/Downloads/breach-v0.3.20-alpha.5-darwin-ia32/Breach.app/Contents/Resources/shell/node_modules/async/lib/async.js:557:34)
    at processImmediate [as _immediateCallback] (timers.js:374:17)
```

and the Network tab in Breach shows me the following response body from the POST:

```
Error: ENOENT, stat '/Users/<username>/mod_strip'
    at Error (native)
```

The POST request payload is the following JSON string:

```
{ "path": "local:/Users/<username>/mod_strip" }
```

Edit 3: If I change the headers in the Postman request to match those of the request within Breach, I also get the same error as I got in Breach:

```
Error: ENOENT, stat '/Users/<username>/mod_strip' at Error (native)
```

**Comment by AdrianTP** [comment]
Ok, long story short, I'm tired and should not try to do work while tired. I cloned mod_strip in the parent directory without realising it. I forgot to `cd` into the directory before cloning the repo. Always check the "obvious" things first. Figuring this out required running the Node REPL and doing fs.stat() on every directory leading up to where I thought I had cloned mod_strip. Once I got there and saw it fail, I decided to check permissions in Bash. `ls -la` displayed no folders, but `cd`ing up a directory and `ls -la` again showed me my folly.
